File: LinkedListStuff.cs
using System;

public class Program
{
	public static void Main()
	{
		Node root = new Node();
		root.data = 1;
		root.next = null;
		
		Node child1 = new Node();
		child1.data = 2;
		child1.next = null;
		
		root.next = child1;
		
		Node child2 = new Node();
		child2.data = 3;
		child2.next = null;
		
		child1.next = child2;
		
		root = reverseLL( root );
		
		printLL( root );
	}
	
	public static void printLL( Node n )
	{
		Console.WriteLine( n.data );
		if (n.next != null)
			printLL( n.next );		
	}
	
	public static Node reverseLL( Node n )
	{
		if (n == null || n.next == null)
		{
			return n;
		}
		
		Node ret = reverseLL( n.next );
		
		Node nextNode = n.next;
		n.next = null;
		nextNode.next = n;
		
		return ret;
	}			
}

public class Node 
{
	public int data;
	public Node next;
}

File: LinkedListStuff.js
var Node = function( value, next ) {
    return {
        value: value,
        next: next
    }
}

var printListForward = function( list ) {
    if (list == null) {
        return; 
    }
    document.getElementById('output').innerHTML += '<li>' + list.value + '</li>';
    return printListForward( list.next );
}

var printListReverse = function( list ) {
    if (list == null) {
        return;
    }
    var ret = printListReverse( list.next );
    document.getElementById('output').innerHTML += '<li>' + list.value + '</li>';
    return ret;
}

var reverseList = function( list ) {
    if (list == null || list.next == null) {
           return list;
    }
    var ret = reverseList( list.next );
    
    var nextNode = list.next;
    list.next = null;
    nextNode.next = list;
    return ret;
}

var list = Node(1, Node(2, Node(3)));
//printListReverse( list );
list = reverseList( list );
printListForward( list );
